python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+            list of subgraphs nodes, and list of pruned nodes.
+
+        """
+        subgraphs = []
+        subgraphs_nodes_mapping = []
+
+        out_degrees = graph.out_degrees()
+        in_degrees = graph.in_degrees()
+        nodes = graph.nodes()
+        node_degree = []
+
+        for node, in_degree, out_degree in zip(nodes, in_degrees, out_degrees):
+            node_degree.append((node, in_degree + out_degree))
+
+        node_degree = sorted(
+            node_degree, key=lambda x: x[1], reverse=bool(high2low)
+        )
+
+        if len(node_degree) < self.num_child_expand:
+            pruned_nodes = nodes
+        else:
+            pruned_nodes = [
+                node_degree[i][0] for i in range(self.num_child_expand)
+            ]
+
+        for prune_node in pruned_nodes:
+            new_subgraph_nodes = np.array(
+                [node for node in nodes if node != prune_node]
+            )
+            new_subgraph = dgl.node_subgraph(graph, new_subgraph_nodes)
+            biggest_comp = self.biggest_weak_component(new_subgraph)
+            new_subgraph_big_comp = dgl.node_subgraph(graph, biggest_comp)
+
+            subgraphs_nodes_mapping.append(new_subgraph_nodes[biggest_comp])
+            subgraphs.append(new_subgraph_big_comp)
+
+        return subgraphs, subgraphs_nodes_mapping, pruned_nodes
+
+    def explain_graph(self, graph, feat, **kwargs):
+        r"""Find the subgraph that play a crucial role to explain the prediction made
+        by the GNN for a graph.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+        feat: Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        kwargs: dict
+            Additional arguments passed to the GNN model. Tensors whose
+            first dimension is the number of nodes or edges will be
+            assumed to be node/edge features.
+
+        Returns
+        -------
+        list
+            Return list of nodes that represent the most important subgraph.
+
+        Examples
+        --------
+
+        >>> import dgl.function as fn
+        >>> import torch
+        >>> import torch.nn as nn
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import AvgPooling, SubgraphXExplainer
+
+        >>> # Load dataset
+        >>> data = GINDataset('MUTAG', self_loop=True)
+        >>> dataloader = GraphDataLoader(data, batch_size=64, shuffle=True)
+
+        >>> # Define a model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super(Model, self).__init__()
+        ...         self.linear = nn.Linear(in_feats, out_feats)
+        ...         self.pool = AvgPooling()
+        ...
+        ...     def forward(self, graph, feat, eweight=None):
```

Comment:
Use a built-in NN module, say `GraphConv` or `SAGEConv` to simplify the implementation.

Reply:
done.